plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -180,14 +183,17 @@ export async function createRouter(
       });
       res.json(actionsList);
     })
-    .post('/v2/tasks', async (req, res) => {
+    .post('/v2/tasks', async (req: AuthenticatedBackstageRequest, res) => {
       const templateRef: string = req.body.templateRef;
       const { kind, namespace, name } = parseEntityRef(templateRef, {
         defaultKind: 'template',
       });
-      const { token, entityRef: userEntityRef } = parseBearerToken(
-        req.headers.authorization,
-      );
+      const token =
+        req.backstage?.identity?.token ||
+        parseBearerToken(req.headers.authorization).token;
+      const userEntityRef =
+        req.backstage?.identity?.identity?.userEntityRef ||
+        parseBearerToken(req.headers.authorization).entityRef;
```

Comment:
Maybe out of scope of this change specifically but my understanding was that the new pattern for obtaining/verifying the (default jwt) user token is:
```ts
const token = getBearerTokenFromAuthorizationHeader(req.header('authorization'));
  
const user = await identity.authenticate(token);
```

Reply:
That looks like a better pattern. Let me implement a provider that does this. I would be inclinded to delete the parseBearerToken function in the scaffolder but I would be afraid to break anyone. I left it here to give people time to uptake the middleware.